packages/modules/fulfillment/src/services/fulfillment-module-service.ts

```diff
@@ -997,6 +1138,32 @@ export default class FulfillmentModuleService<
       sharedContext
     )
 
+    buildServiceZoneEvents({
```

Comment:
Is the decorator necessary, or should we just do `emitServiceZoneEvents`? What does the decorator do?

Reply:
The decorator inject the MessageAggregator and wrap the function such that the emit is called after the function execution. In those utils I never emit anything, otherwise we could emit things within a running transaction and then the transaction fail